Helpers/MarkerManager.m

```diff
@@ -0,0 +1,142 @@
+//
+//  MarkerManager.m
+//  social-maps
+//
+//  Created by Bevin Benson on 7/27/18.
+//  Copyright © 2018 Bevin Benson. All rights reserved.
+//
+
+#import "MarkerManager.h"
+#import "Marker.h"
+
+NSString *const kFavoritesKey = @"favorites";
+NSString *const kWishlistKey = @"wishlist";
+NSString *const kFollowFavKey = @"followFavorites";
+
+@implementation MarkerManager
++ (instancetype)shared {
+    static MarkerManager* sharedManager = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedManager = [[self alloc] init];
+    });
+    
+    return sharedManager;
+}
+
+- (void)initMarkerDictionaries {
+    self.markersByMarkerType = [NSMutableDictionary new];
+    [self.markersByMarkerType setObject:[[NSMutableArray alloc] init] forKey:kFavoritesKey];
+    [self.markersByMarkerType setObject:[[NSMutableArray alloc] init] forKey:kWishlistKey];
+    [self.markersByMarkerType setObject:[[NSMutableArray alloc] init] forKey:kFollowFavKey];
+    
+    self.typeDict = @{
+                      @"Entertainment":@[@"amusement_park", @"aquarium", @"casino", @"movie_theater", @"bowling_alley", @"zoo", @"night_club"],
+                      @"Restaurants":@[@"restaurant", @"meal_delivery", @"meal_takeaway"], @"Café":@[@"cafe", @"bakery"],
+                      @"Shopping":@[@"clothing_store", @"department_store", @"home_goods_store", @"shopping_mall", @"shoe_store", @"furniture_store"],
+                      @"Outdoors":@[@"park", @"campground", @"rv_park"],
+                      @"Beauty":@[@"beauty_salon", @"hair_care"],
+                      @"Museums":@[@"art_gallery", @"museum"]
+                      };
+    
+    self.placeCategories = [self.typeDict allKeys];
+    self.markersByPlaceCategory = [NSMutableDictionary new];
+    for (NSString *key in self.placeCategories) {
+        [self.markersByPlaceCategory setObject:[NSMutableArray new] forKey:key];
+    }
+    
+    // Reverse typeDict so that each Google Place type is mapped to its category name
+    NSMutableDictionary *mutableDetailedTypeDict = [NSMutableDictionary new];
```

Comment:
as we chatted about, refactoring this to use anotehr helper class could be a good exercise :)